src/Payments/PaymentMethodRegistry.php

```diff
@@ -17,60 +17,46 @@ final class PaymentMethodRegistry {
 	/**
 	 * Registered payment methods, as `$name => $instance` pairs.
 	 *
-	 * @var PaymentMethodType[]
+	 * @var PaymentMethodTypeInterface[]
 	 */
 	private $registered_payment_methods = [];
 
-	/**
-	 * Container for the main instance of the class.
-	 *
-	 * @var PaymentMethodRegistry|null
-	 */
-	private static $instance = null;
-
 	/**
 	 * Registers a payment method.
 	 *
-	 * @param string|PaymentMethodType $name Payment method name, or alternatively a PaymentMethodType instance.
-	 * @param array                    $args Optional. Array of payment method type arguments.
-	 * @return PaymentMethodType|false The registered payment method type on success, or false on failure.
+	 * @param PaymentMethodTypeInterface $payment_method_type An instance of PaymentMethodTypeInterface.
+	 *
+	 * @return boolean True means registered successfully.
 	 */
-	public function register( $name, $args = array() ) {
-		$payment_method = null;
-
-		if ( $name instanceof PaymentMethodType ) {
-			$payment_method = $name;
-			$name           = $payment_method->name;
-		}
-
-		if ( ! is_string( $name ) ) {
-			_doing_it_wrong( __METHOD__, esc_html__( 'Payment method names must be strings.', 'woo-gutenberg-products-block' ), '2.5.0' );
-			return false;
-		}
-
+	public function register( PaymentMethodTypeInterface $payment_method_type ) {
+		$name = $payment_method_type->get_name();
 		if ( $this->is_registered( $name ) ) {
 			/* translators: %s: Payment method name. */
 			_doing_it_wrong( __METHOD__, esc_html( sprintf( __( 'Payment method "%s" is already registered.', 'woo-gutenberg-products-block' ), $name ) ), '2.5.0' );
 			return false;
 		}
+		$this->registered_payment_methods[ $name ] = $payment_method_type;
+		return true;
+	}
 
-		if ( ! $payment_method ) {
-			$payment_method = new PaymentMethodType( $name, $args );
+	/**
+	 * Initializes all payment method types.
+	 */
+	public function initialize() {
+		$registered_payment_method_types = $this->get_all_registered();
+		foreach ( $registered_payment_method_types as $registered_type ) {
+			$registered_type->initialize();
 		}
```

Comment:
One initialization method to handle initializing all registered payment methods.